src/generic_cube.rs

```diff
@@ -185,13 +185,56 @@ pub enum Move {
     Z(MoveVariant),
 }
 
+impl Move {
+    /// Extracts the MoveVariant of a Move.
+    pub fn get_variant(&self) -> MoveVariant {
+        match self {
+            Move::U(v)
+            | Move::L(v)
+            | Move::F(v)
+            | Move::R(v)
+            | Move::B(v)
+            | Move::D(v)
+            | Move::X(v)
+            | Move::Y(v)
+            | Move::Z(v)
+            | Move::Uw(_, v)
+            | Move::Lw(_, v)
+            | Move::Fw(_, v)
+            | Move::Rw(_, v)
+            | Move::Bw(_, v)
+            | Move::Dw(_, v) => *v,
+        }
+    }
+
+    /// Returns the Move with the given MoveVariant.
+    pub fn with_variant(&self, variant: MoveVariant) -> Move {
```

Comment:
This is acceptable, but there might be a better way to do this. I'll come back to this with comments later.

Reply:
Other than restructuring the Move enum, there isn't.